Compile bug: redefinition of 'res_error' as different kind of symbol

### Git commit

current master

### Operating systems

Linux

### GGML backends

CUDA

### Problem description & steps to reproduce

compile llama.cpp using toolchains from anaconda

### First Bad Commit

ec18edfcba94dacb166e6523612fc0129cead67a

### Compile command

```shell
cmake --build llama.cpp/build --config Release -j 16 --target llama-server
```

### Relevant log output

```shell
[1/2] Building CXX object tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o
FAILED: tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o 
/work/miniforge3/envs/cuda13/bin/clang++ -DGGML_USE_CPU -DGGML_USE_CUDA -DGGML_USE_RPC -DLLAMA_USE_HTTPLIB -I/home/builduser/llama.cpp/tools/server -I/home/builduser/llama.cpp/build/tools/server -I/home/builduser/llama.cpp/tools/server/../mtmd -I/home/builduser/llama.cpp -I/home/builduser/llama.cpp/common/. -I/home/builduser/llama.cpp/common/../vendor -I/home/builduser/llama.cpp/src/../include -I/home/builduser/llama.cpp/ggml/src/../include -I/home/builduser/llama.cpp/tools/mtmd/. -fPIC -static-libgcc -static-libstdc++ -static-libgfortran -g0 -O3 -march=native -mtune=native -feliminate-unused-debug-types -pipe -Wall -Wno-unused-command-line-argument -fasynchronous-unwind-tables -Wl,-z,now -Wl,-z,relro -fno-semantic-interposition -fno-fat-lto-objects -fno-trapping-math -Wl,-sort-common -Wl,--enable-new-dtags -ffunction-sections -fvisibility-inlines-hidden -Wl,--enable-new-dtags -flto=full -fuse-ld=lld -O3 -DNDEBUG -flto=thin -fPIE -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wunreachable-code-break -Wunreachable-code-return -Wmissing-prototypes -Wextra-semi -pthread -MD -MT tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o -MF tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o.d -o tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o -c /home/builduser/llama.cpp/tools/server/server-models.cpp
/home/builduser/llama.cpp/tools/server/server-models.cpp:590:13: error: redefinition of 'res_error' as different kind of symbol
  590 | static void res_error(std::unique_ptr<server_http_res> & res, const json & error_data) {
      |             ^
/work/miniforge3/envs/cuda13/x86_64-conda-linux-gnu/sysroot/usr/include/resolv.h:71:65: note: previous definition is here
   71 | typedef enum { res_goahead, res_nextns, res_modified, res_done, res_error }
      |                                                                 ^
/home/builduser/llama.cpp/tools/server/server-models.cpp:597:18: error: called object type 'res_sendhookact' is not a function or function pointer
  597 |         res_error(res, format_error_response("model name is missing from the request", ERROR_TYPE_INVALID_REQUEST));
      |         ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:602:18: error: called object type 'res_sendhookact' is not a function or function pointer
  602 |         res_error(res, format_error_response("model not found", ERROR_TYPE_INVALID_REQUEST));
      |         ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:609:22: error: called object type 'res_sendhookact' is not a function or function pointer
  609 |             res_error(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:709:22: error: called object type 'res_sendhookact' is not a function or function pointer
  709 |             res_error(res, format_error_response("model is not found", ERROR_TYPE_NOT_FOUND));
      |             ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:713:22: error: called object type 'res_sendhookact' is not a function or function pointer
  713 |             res_error(res, format_error_response("model is already loaded", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:771:22: error: called object type 'res_sendhookact' is not a function or function pointer
  771 |             res_error(res, format_error_response("model is not found", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^
/home/builduser/llama.cpp/tools/server/server-models.cpp:775:22: error: called object type 'res_sendhookact' is not a function or function pointer
  775 |             res_error(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^
8 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/tools/server/server-models.cpp
+++ b/tools/server/server-models.cpp
@@ -587,26 +587,26 @@ static void res_ok(std::unique_ptr<server_http_res> & res, const json & response
     res->data = safe_json_to_str(response_data);
 }
 
-static void res_error(std::unique_ptr<server_http_res> & res, const json & error_data) {
+static void res_err(std::unique_ptr<server_http_res> & res, const json & error_data) {
     res->status = json_value(error_data, "code", 500);
     res->data = safe_json_to_str({{ "error", error_data }});
 }
 
 static bool router_validate_model(const std::string & name, server_models & models, bool models_autoload, std::unique_ptr<server_http_res> & res) {
     if (name.empty()) {
-        res_error(res, format_error_response("model name is missing from the request", ERROR_TYPE_INVALID_REQUEST));
+        res_err(res, format_error_response("model name is missing from the request", ERROR_TYPE_INVALID_REQUEST));
         return false;
     }
     auto meta = models.get_meta(name);
     if (!meta.has_value()) {
-        res_error(res, format_error_response("model not found", ERROR_TYPE_INVALID_REQUEST));
+        res_err(res, format_error_response("model not found", ERROR_TYPE_INVALID_REQUEST));
         return false;
     }
     if (models_autoload) {
         models.ensure_model_loaded(name);
     } else {
         if (meta->status != SERVER_MODEL_STATUS_LOADED) {
-            res_error(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
             return false;
         }
     }
@@ -706,11 +706,11 @@ void server_models_routes::init_routes() {
         std::string name = json_value(body, "model", std::string());
         auto model = models.get_meta(name);
         if (!model.has_value()) {
-            res_error(res, format_error_response("model is not found", ERROR_TYPE_NOT_FOUND));
+            res_err(res, format_error_response("model is not found", ERROR_TYPE_NOT_FOUND));
             return res;
         }
         if (model->status == SERVER_MODEL_STATUS_LOADED) {
-            res_error(res, format_error_response("model is already loaded", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("model is already loaded", ERROR_TYPE_INVALID_REQUEST));
             return res;
         }
         models.load(name, false);
@@ -768,11 +768,11 @@ void server_models_routes::init_routes() {
         std::string name = json_value(body, "model", std::string());
         auto model = models.get_meta(name);
         if (!model.has_value()) {
-            res_error(res, format_error_response("model is not found", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("model is not found", ERROR_TYPE_INVALID_REQUEST));
             return res;
         }
         if (model->status != SERVER_MODEL_STATUS_LOADED) {
-            res_error(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("model is not loaded", ERROR_TYPE_INVALID_REQUEST));
             return res;
         }
         models.unload(name);

